taranis/analyze_schema.py

```diff
@@ -0,0 +1,226 @@
+import logging
+import pandas as pd
+import os
+import rich.console
+import statistics
+from pathlib import Path
+
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+
+import taranis.utils
+
+import pdb
+
+log = logging.getLogger(__name__)
+stderr = rich.console.Console(
+    stderr=True,
+    style="dim",
+    highlight=False,
+    force_terminal=taranis.utils.rich_force_colors(),
+)
+
+
+class AnalyzeSchema:
+    def __init__(
+        self,
+        schema_allele,
+        output,
+        remove_subset,
+        remove_duplicated,
+        remove_no_cds,
+        genus,
+        species,
+        usegenus,
+    ):
+        self.schema_allele = schema_allele
+        self.allele_name = Path(self.schema_allele).stem
+        self.output = output
+        self.remove_subset = remove_subset
+        self.remove_duplicated = remove_duplicated
+        self.remove_no_cds = remove_no_cds
+        self.genus = genus
+        self.species = species
+        self.usegenus = usegenus
+
+    def check_allele_quality(self):
+        a_quality = {}
+        allele_seq = {}
+        bad_quality_record = []
+        with open(self.schema_allele) as fh:
+            for record in SeqIO.parse(self.schema_allele, "fasta"):
+                a_quality[record.id] = {"quality": "Good quality", "reason": "-"}
+                allele_seq[record.id] = str(record.seq)
+                a_quality[record.id]["length"] = len(str(record.seq))
+                if len(record.seq) % 3 != 0:
+                    a_quality[record.id]["quality"] = "Bad quality"
+                    a_quality[record.id]["reason"] = "Can not be converted to protein"
```

Comment:
I would call this as in the previous code : Not a CDS, cannot be converted to protein. Or something like this

Reply:
error messages are collected from biopython text message